lib/vmpooler/prometheus.rb

```diff
@@ -0,0 +1,240 @@
+        },
+        checkout: {
+          mtype: M_COUNTER,
+          docstring: 'vmpooler checkout',
+          prom_metric_prefix: 'vmpooler_checkout',
+          metric_suffixes: %i[nonresponsive empty success invalid],
+          param_labels: %i[poolname]
+        },
+        config: {
+          mtype: M_COUNTER,
+          docstring: 'vmpooler configuration',
+          prom_metric_prefix: 'vmpooler_config',
+          metric_suffixes: %i[invalid],  # This is an odd one with only one suffix
+          param_labels: %i[poolname]
+        },
+        poolreset: {
+          mtype: M_COUNTER,
+          docstring: 'vmpooler reset pool',
+          prom_metric_prefix: 'vmpooler_poolreset',
+          metric_suffixes: %i[invalid],  # This is an odd one with only one suffix
+          param_labels: %i[poolname]
+        },
+        connect: {
+          mtype: M_COUNTER,
+          docstring: 'vmpooler connect',
+          prom_metric_prefix: 'vmpooler_connect',
+          metric_suffixes: %i[open fail],
+          param_labels: []
+        },
+        migrate_from: {
+          mtype: M_COUNTER,
+          docstring: 'vmpooler machine migrated from',
+          prom_metric_prefix: 'vmpooler_migrate_from',
+          param_labels: %i[host_name]
+        },
+        migrate_to: {
+          mtype: M_COUNTER,
+          docstring: 'vmpooler machine migrated to',
+          prom_metric_prefix: 'vmpooler_migrate_to',
+          param_labels: %i[host_name]
+        },
+        ready: {
+          mtype: M_GAUGE,
+          docstring: 'vmpooler machines ready',
+          prom_metric_prefix: 'vmpooler_ready',
+          param_labels: %i[poolname]
+        },
+        running: {
+          mtype: M_GAUGE,
+          docstring: 'vmpooler machines running',
+          prom_metric_prefix: 'vmpooler_running',
+          param_labels: %i[poolname]
+        },
+        time_to_ready_state: {
+          mtype: M_HISTOGRAM,
+          docstring: 'vmpooler machine time to ready',
+          prom_metric_prefix: 'vmpooler_time_to_ready_state',
+          param_labels: %i[poolname]
+        },
+        clone: {
+          mtype: M_HISTOGRAM,
+          docstring: 'vmpooler machine clone time',
+          prom_metric_prefix: 'vmpooler_clone',
+          param_labels: %i[poolname]
+        },
+        destroy: {
+          mtype: M_HISTOGRAM,
+          docstring: 'vmpooler machine destroy time',
+          prom_metric_prefix: 'vmpooler_destroy',
+          param_labels: %i[poolname]
+        },
+        connection_available: {
+          mtype: M_GAUGE,
+          docstring: 'connection available count',
+          prom_metric_prefix: 'vmpooler_connection_available',
+          param_labels: %i[provider]
+        },
+        connection_waited: {
+          mtype: M_HISTOGRAM,
+          docstring: 'connection wait time',
+          prom_metric_prefix: 'vmpooler_connection_waited',
+          param_labels: %i[provider]
+        }
+      }
+    end
```

Comment:
General feedback about the hash above: would it be practical to make the doc strings a little more detailed? Those strings are a big help to someone looking at metrics who is not an expert on the inner workings of the application.

Reply:
TBD